### DISCUSSION TITLE
How to download and use ksud command on Windows

### DISCUSSION THREAD

**Discussion by Dondrejohnson5** [tiann/KernelSU#1538] [Q&A] [answered]
Hello, I would like to use the ksud command to patch boot imgs with KLM, but the instructions are very vague on how to even install the commands to use on a platform like Windows. Can I please get some guidance? I would very much appreciate it.

**Comment by tiann** [comment, marked as answer]
You can download it here: https://github.com/tiann/KernelSU/actions/workflows/build-manager.yml

**Reply by schnatterer** [reply]
Is there a reason it's not appended to the releases (any more)? This makes it quite hard to find our use in downstream projects.

**Reply by schnatterer** [reply]
This works, but is clumsy and requires authentication at GitHub.

```bash
KERNELSU_VERSION='v0.9.3'
GITHUB_TOKEN='...'

local artifacts_url ksud_artifact_api_url 

artifacts_url=$(curl -s --fail -H "Authorization: token $GITHUB_TOKEN" \
  "https://api.github.com/repos/tiann/KernelSU/actions/runs?event=push&branch=$KERNELSU_VERSION" | jq  -r '.workflow_runs[0].artifacts_url')
# Note that this might fail once we exceed 100 artifacts. Then, we would have to implement paging
ksud_artifact_api_url=$(curl -s --fail -H "Authorization: token $GITHUB_TOKEN" "$artifacts_url?per_page=100" \
  | jq -r '.artifacts[] | select(.name == "ksud-x86_64-unknown-linux-musl" )| .archive_download_url')
 curl --fail -Lo ".tmp/ksud-$KERNELSU_VERSION.zip" -H "Authorization: token $GITHUB_TOKEN" "$ksud_artifact_api_url"
 # Note that for this authentication is required!
 # Even this 👇️ returns 404 with curl, works in browser only when logged in
 # run_id=$(echo $artifacts_url | sed -n 's/.*runs\/\([0-9]*\)\/.*/\1/p')
 # ksud_artifact_id=$(echo $ksud_artifact_api_url | sed -n 's/.*artifacts\/\([0-9]*\)\/.*/\1/p')
 # curl --fail -Lo ".tmp/ksud.zip" "https://github.com/tiann/KernelSU/actions/runs/$run_id/artifacts/$ksud_artifact_id"
```

**Reply by gricn** [reply]
> You can download it here: https://github.com/tiann/KernelSU/actions/workflows/build-manager.yml

Thanks for your reply but I still don't understand how to use it. I found a [ksud.yml] file(https://github.com/tiann/KernelSU/blob/main/.github/workflows/ksud.yml) in the github workflow and it seems a little bit different from the code @schnatterer  provide there.

So why couldn't we provide the executive file in the Releases Assets or extract it to another repository?
